File: admin/src/api/schedule.ts
import { request } from './client'

export interface ScheduleInput {
  doctor_id: number
  schedule_date: string
  time_slot: string
  total_slots: number
}

export interface Schedule extends ScheduleInput {
  id: number
  remaining_slots: number
  is_active: boolean
}

export const scheduleApi = {
  list: () => request<Schedule[]>('/b/schedules'),
  create: (payload: ScheduleInput) => request<Schedule>('/b/schedules', {
    method: 'POST',
    body: JSON.stringify(payload),
  }),
  disable: (id: number) => request<Schedule>(`/b/schedules/${id}/disable`, {
    method: 'PATCH',
  }),
}

File: server/app/schemas/schedule.py
from datetime import date

from pydantic import BaseModel, ConfigDict, Field

from app.models.schedule import TimeSlot


class ScheduleInput(BaseModel):
    doctor_id: int
    schedule_date: date
    time_slot: TimeSlot
    total_slots: int = Field(gt=0)


class ScheduleResponse(ScheduleInput):
    model_config = ConfigDict(from_attributes=True)

    id: int
    remaining_slots: int
    is_active: bool
